tripy/tests/flat_ir/ops/test_iota.py

```diff
@@ -49,9 +49,11 @@ def test_mlir(self, flat_ir):
             """
             module {
                 func.func @main() -> tensor<?x?xf32> {
-                    %c = stablehlo.constant dense<[2, 3]> : tensor<2xi32>
-                    %0 = stablehlo.dynamic_iota %c, dim = 0 : (tensor<2xi32>) -> tensor<?x?xf32>
-                    return %0 : tensor<?x?xf32>
+                    %c = stablehlo.constant dense<2> : tensor<1xi32>
+                    %c_0 = stablehlo.constant dense<3> : tensor<1xi32>
+                    %0 = stablehlo.concatenate %c, %c_0, dim = 0 : (tensor<1xi32>, tensor<1xi32>) -> tensor<2xi32>
+                    %1 = stablehlo.dynamic_iota %0, dim = 0 : (tensor<2xi32>) -> tensor<?x?xf32>
+                    return %1 : tensor<?x?xf32>
```

Comment:
What is the reason for this change?

Reply:
Perhaps a consequence of having lists of dimensions rather than single shape tensors?

Reply:
I wonder if the same IR bloat will be present once we move to the tensorrt dialect.

Reply:
I did not understand the difference, the original one should just work.

Reply:
Yes, it's a consequence of having lists of dimensions. `Shape([2, 3])` will materialize the two element tensor, whereas now we are doing a `concat(DimensionSize([2]), DimensionSize([3]))`.

Reply:
This is resolved by the fix for: https://github.com/NVIDIA/TensorRT-Incubator/pull/306#discussion_r1821340540
The IR should now be the same as before.